core/commands/storage.go

```diff
@@ -291,16 +293,18 @@ func controlSessionTimeout(ss *storage.Session) {
 	}
 }
 
-func retryMonitor(ctx context.Context, api coreiface.CoreAPI, ss *storage.Session, n *core.IpfsNode, ssID string, test bool) {
+func retryMonitor(ctx context.Context, api coreiface.CoreAPI, ss *storage.FileContracts, n *core.IpfsNode, ssID string, test bool) {
 	retryQueue := ss.GetRetryQueue()
 	if retryQueue == nil {
 		log.Error("retry queue is nil")
 		return
 	}
 
 	// loop over each chunk
-	for chunkHash, chunkInfo := range ss.ChunkInfo {
-		go func(chunkHash string, chunkInfo *storage.Chunk) {
+	shardIndex := 0
+	for chunkHash, chunkInfo := range ss.ShardInfo {
+		shardIndex++
```

Comment:
`shardIndex` starts from 1 not 0? Is this a guard requirement or just a locally stored index?